shell/common/shell.cc

```diff
@@ -706,29 +706,23 @@ void Shell::OnPlatformViewRegisterTexture(
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
-  task_runners_.GetGPUTaskRunner()->PostTask(
-      [rasterizer = rasterizer_->GetWeakPtr(), texture] {
-        if (rasterizer) {
-          if (auto* registry = rasterizer->GetTextureRegistry()) {
-            registry->RegisterTexture(texture);
-          }
-        }
-      });
+  if (rasterizer_) {
+    if (auto* registry = rasterizer_->GetTextureRegistry()) {
+      registry->RegisterTexture(texture);
+    }
+  }
 }
 
 // |PlatformView::Delegate|
 void Shell::OnPlatformViewUnregisterTexture(int64_t texture_id) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
-  task_runners_.GetGPUTaskRunner()->PostTask(
-      [rasterizer = rasterizer_->GetWeakPtr(), texture_id]() {
-        if (rasterizer) {
-          if (auto* registry = rasterizer->GetTextureRegistry()) {
-            registry->UnregisterTexture(texture_id);
-          }
-        }
-      });
+  if (rasterizer_) {
+    if (auto* registry = rasterizer_->GetTextureRegistry()) {
```

Comment:
The rasterizer and the texture registry is only safe to access on the GPU task runner. This call happens on the platform task runner. This is not thread safe.

Reply:
You are right, the patch is not thread safe. There is another solution, we can avoiding the object implementing `FlutterTexture` protocol  to be released too early by simply retaining it in engine.

Reply:
@chinmaygarde I have made changes.